### ISSUE TITLE
Strange gap when using mdc-toolbar--flexible

### ISSUE THREAD

**Issue by ocavue** [material-components/material-components-web#1832]
### What MDC-Web Version are you using?

> 0.27.0 (not very sure)

### What browser(s) is this bug affecting?

> Chrome 63.0

### What OS are you using?

> macOS 10.3

### What are the steps to reproduce the bug?

using phone (easiest way):
1. find a phone with a 1080p screen (iPhone 6s plus and google pixel for example)
2. set phone to horizontal
3. open [this url](https://material-components-web.appspot.com/toolbar/waterfall-flexible-toolbar.html), which is a demo page for waterfall-flexible-toolbar created by Google.
4. scroll page to the bottom

using computer:
1. open link above
2. change your browser's size: width=610px, height=400px. (Open inspect so chrome will show window size)
3. reload page
4. scroll page to the bottom

### What is the expected behavior?

All items in toolbar are aligned

### What is the actual behavior?

(see Screen Shot below)
the height of toolbar itself is 64px;
the height of icons and title are 56px (include padding);
all icons are on the top of toolbar and title are on the bottom of the toolbar;

if you are using phone:
![phone](https://user-images.githubusercontent.com/24715727/34319856-1919c906-e827-11e7-9cac-76d2da3d6dab.jpg)

if you are using computer:
![screen shot 2017-12-23 at 21 02 50 2](https://user-images.githubusercontent.com/24715727/34319764-f012427e-e824-11e7-965a-72ab2a6dd97a.png)
![screen shot 2017-12-23 at 21 05 38](https://user-images.githubusercontent.com/24715727/34319782-33c90b60-e825-11e7-82ad-ccd8587e4956.png)
![screen shot 2017-12-23 at 21 05 58](https://user-images.githubusercontent.com/24715727/34319784-345e43ec-e825-11e7-8d8e-18336ede490f.png)
![screen shot 2017-12-23 at 21 05 46](https://user-images.githubusercontent.com/24715727/34319783-3411425e-e825-11e7-9dd0-7e56d9e8f984.png)

### Any other information you believe would be useful?

I am not sure if it's a bug or a feature. 
look forward to your replying.

**Comment by acdvorak** [comment]
Yep, that's a bug! Thanks for taking the time to file this issue! 😄

**Comment by ocavue** [comment]
@acdvorak After Google I/O 2018, I found that MDC-web's demo has been moved from https://material-components-web.appspot.com to https://material-components.github.io. In the new site, [Top App Bar](https://material-components.github.io/material-components-web-catalog/#/component/top-app-bar) has not "waterfall-flexible-toolbar" anymore. Does google give up "waterfall-flexible-toolbar"? If so, this issue could be closed.

**Comment by kfranqueiro** [comment]
Thanks for checking in on this issue.

You're right; there aren't immediate plans to implement the waterfall flexible toolbar in Top App Bar, and MDC Toolbar as it currently exists is being deprecated, so for the time being it makes sense to close this. If we do implement a waterfall flexible variant for Top App Bar in the future, it will most likely be redesigned such that this issue won't occur.
